From 55b9bb170f20144160bf1a1d5c99e5ed59f59cf4 Mon Sep 17 00:00:00 2001
From: Cyril Arnould <cyril.arnould@outlook.com>
Date: Wed, 14 Jun 2023 00:06:38 +0200
Subject: [PATCH 12/17] Skip drop_privs entirely in mingw variant

drop_privs wasn't implemented in ezwinports so it's not here either
---
 locate/locate.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/locate/locate.c b/locate/locate.c
index 9fda81bd..ddc718c9 100644
--- a/locate/locate.c
+++ b/locate/locate.c
@@ -1440,6 +1440,7 @@ static struct option const longopts[] =
 static int
 drop_privs (void)
 {
+#if !defined(__MINGW32__) && !defined(_MSC_VER)
   const char * what = "failed";
   const uid_t orig_euid = geteuid ();
   const uid_t uid       = getuid ();
@@ -1523,6 +1524,7 @@ drop_privs (void)
     {
       /* deliberate infinite loop */
     }
+#endif
 }
 
 static int
-- 
2.41.0

